newrelic/resource_newrelic_application.go

```diff
@@ -0,0 +1,130 @@
+package newrelic
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/newrelic/newrelic-client-go/pkg/apm"
+)
+
+func resourceNewRelicApplication() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNewRelicApplicationCreate,
+		Read:   resourceNewRelicApplicationRead,
+		Update: resourceNewRelicApplicationUpdate,
+		Delete: resourceNewRelicApplicationDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"app_apdex_threshold": {
+				Type:     schema.TypeFloat,
+				Required: true,
+			},
+			"end_user_apdex_threshold": {
+				Type:     schema.TypeFloat,
+				Required: true,
+			},
+			"enable_real_user_monitoring": {
+				Type:     schema.TypeBool,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceNewRelicApplicationCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ProviderConfig).NewClient
+
+	userApp := expandApplication(d)
+	log.Printf("[ZACH] CREATE THING %+v\n\n\n\n", userApp)
+
+	listParams := apm.ListApplicationsParams{
+		Name: userApp.Name,
+	}
+
+	result, err := client.APM.ListApplications(&listParams)
+	if err != nil {
+		return err
+	}
+
+	if len(result) != 1 {
+		return fmt.Errorf("more/less than one result from query for %s", userApp.Name)
+	}
+
+	app := *result[0]
+
+	if app.Name != userApp.Name {
+		return fmt.Errorf("the result name %s does not match requested name %s", app.Name, userApp.Name)
+	}
+
+	d.SetId(strconv.Itoa(app.ID))
+
+	log.Printf("[INFO] Importing New Relic application %v", userApp.Name)
+	return resourceNewRelicApplicationUpdate(d, meta)
+}
+
+func resourceNewRelicApplicationRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ProviderConfig).NewClient
+
+	userApp := expandApplication(d)
+	log.Printf("[INFO] Reading New Relic application %+v", userApp)
+
+	app, err := client.APM.GetApplication(userApp.ID)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[INFO] Read found New Relic application %+v\n\n\n", app)
+
+	// a := expandApplication(d)
+	// log.Printf("[INFO] Read before flatten  %+v\n\n\n", a)
+
+	flattenApplication(app, d)
```

Comment:
The `d` here looks the same before and after the flatten, so I'm a little unclear what this is doing.